Firestore/Swift/Source/Codable/DocumentSnapshot+ReadDecodable.swift

```diff
@@ -35,11 +35,9 @@ extension DocumentSnapshot {
   ///     default decoder.
   public func data<T: Decodable>(as type: T.Type,
                                  with serverTimestampBehavior: ServerTimestampBehavior = .none,
-                                 decoder: Firestore.Decoder? = nil) throws -> T? {
+                                 decoder: Firestore.Decoder? = nil) throws -> T {
```

Comment:
This is a breaking change. I compiled my FirestoreCodableSamples app against this PR, and every single `snapshot..data(as: )` now breaks:


![CleanShot 2022-01-13 at 20 29 44@2x](https://user-images.githubusercontent.com/232107/149396687-3c02b76a-b0b6-4749-8b52-5348557de6cd.png)

Reply:
Right - should I try going for keeping the old api with a deprecation warning - and then see if I can come up with a new name for the replacement?
Any suggestions are very welcome. :-)